docs/upgrading.asciidoc

```diff
@@ -37,3 +42,90 @@ The table below is a simplified description of this policy.
 |2.0.x |2020-05-14 |2.1.0
 |1.0.x |2020-02-23 |3.0.0
 |====
+
+[[upgrade-to-v5]]
+=== Upgrade to version 5.x
+
+Upgrading from version `4.x` to `5.x` of the RUM Agent introduces some breaking changes.
+This guide outlines the changes and the steps required to upgrade.
+
+[float]
+[[v5-breaking-changes]]
+=== 5.0 Breaking changes
+
+*Agent name changed to `rum-js`*
+
+The Agent name has been changed to `rum-js`.
+Because of this change, you may need to upgrade your APM Server instance,
+and therefor your Elasticsearch and Kibana instance.
+
+*Official W3C tracecontext support*
+
+The RUM Agent supports the official W3C tracecontext `traceparent` header,
+instead of the previously used `elastic-apm-traceparent` header.
+If you're using Elastic backend agents,
+you must upgrade them to a version that also supports the official W3C tracecontext headers.
+
+*Consolidated configurations*
+
+A single queue is now used to process all events (transactions, errors, etc.).
+This change allows the consolidation of four configuration options into one:
+
+Removed options:
+
+* `errorThrottleLimit`
+* `errorThrottleInterval`
+* `transactionThrottleLimit`
+* `transactionThrottleInterval`
+
+Added option:
+
+* <<event-limit>> -- Configure the number of events sent sent to APM Server per minute. Defaults to `80`.
+
+[float]
+[[v5-upgrade-steps]]
+=== Upgrade steps
+
+[float]
+[[v5-upgrade-server]]
+==== Upgrade APM Server
+
+Version `5.x` of the RUM Agent requires APM Server version >= `7.0`.
+The {apm-server-ref-v}/upgrading-to-70.html[APM Server `7.0` upgrade guide] can help with the upgrade process.
+
+NOTE: APM Server version >= `7.0` requires Elasticsearch and Kibana versions >= `7.0` as well.
+
+[float]
+[[v5-upgrade-agents]]
+==== Upgrade backend agents
+
+All Elastic APM agents have been upgraded to support the changes in the RUM Agent.
+You must upgrade your backend agents to the minimum versions listed below for all features to work:
+
+[options="header"]
+|====
+|Agent name |Agent Version
+|**Go Agent**|>= `1.6`
+|**Java Agent**|>= `1.14`
+|**.NET Agent**|>= `1.3`
+|**Node.js Agent**|>= `3.4`
+|**Python Agent**|>= `5.4`
+|**Ruby Agent**|>= `3.5`
+|====
```

Comment:
Should we instead link to the compatibility table in distributed tracing, or do you think it's fine to have duplicate?

Reply:
Great question. Eventually (v6?) I envision the compatibility table in the DT guide going away. I'd like to single source the guide, but I can't until that PR is merged... OH SNAP, looks like it was. I'll rebase and make a change that single sources the compatibility table.